test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Export/ExportDataTests.cs

```diff
@@ -0,0 +1,225 @@
+
+            IList<Uri> blobUris = await ExportTestHelper.CheckExportStatus(_testFhirClient, contentLocation);
+
+            // Download exported data from storage account
+            Dictionary<(string resourceType, string resourceId, string versionId), Resource> dataFromExport =
+                await ExportTestHelper.DownloadBlobAndParse(blobUris, _fhirJsonParser, _outputHelper);
+
+            // Assert both data are equal
+            Assert.True(ExportTestHelper.ValidateDataFromBothSources(_fixture.TestResources, dataFromExport, _outputHelper));
+            Assert.True(blobUris.All((url) => url.OriginalString.Contains(testContainer)));
+        }
+
+        [Fact]
+        [Trait(Traits.Category, Categories.ExportLongRunning)]
+        public async Task GivenFhirServer_WhenDataIsExportedWithHistoryParallel_ThenExportedDataIsSameAsDataInFhirServer()
+        {
+            await ExportAndSoftDeleteTestHelper(parallel: true, history: true, deletes: false);
+        }
+
+        [Fact]
+        [Trait(Traits.Category, Categories.ExportLongRunning)]
+        public async Task GivenFhirServer_WhenDataIsExportedWithHistoryNotParallel_ThenExportedDataIsSameAsDataInFhirServer()
+        {
+            await ExportAndSoftDeleteTestHelper(parallel: false, history: true, deletes: false);
+        }
+
+        [Fact]
+        [Trait(Traits.Category, Categories.ExportLongRunning)]
+        public async Task GivenFhirServer_WhenDataIsExportedWithSoftDeletesParallel_ThenExportedDataIsSameAsDataInFhirServer()
+        {
+            await ExportAndSoftDeleteTestHelper(parallel: true, history: false, deletes: true);
+        }
+
+        [Fact]
+        [Trait(Traits.Category, Categories.ExportLongRunning)]
+        public async Task GivenFhirServer_WhenDataIsExportedWithSoftDeletesNotParallel_ThenExportedDataIsSameAsDataInFhirServer()
+        {
+            await ExportAndSoftDeleteTestHelper(parallel: false, history: false, deletes: true);
+        }
+
+        [Fact]
+        public async Task GivenFhirServer_WhenDataIsExportedWithHistoryAndSoftDeletesParallel_ThenExportedDataIsSameAsDataInFhirServer()
+        {
+            await ExportAndSoftDeleteTestHelper(parallel: true, history: true, deletes: true);
+        }
+
+        [Fact]
+        [Trait(Traits.Category, Categories.ExportLongRunning)]
+        public async Task GivenFhirServer_WhenDataIsExportedWithHistoryAndSoftDeletesNotParallel_ThenExportedDataIsSameAsDataInFhirServer()
+        {
+            await ExportAndSoftDeleteTestHelper(parallel: false, history: true, deletes: true);
+        }
+
+        // _tag filter cannot be used with history or deleted export. Using isParallel to test both SQL code paths.
+        private async Task ExportAndSoftDeleteTestHelper(bool parallel, bool history, bool deletes)
+        {
+            if (_fixture.DataStore == DataStore.CosmosDb && parallel)
```

Comment:
Instead of checking this you can give the tests the attribute to only run on SQL.
[HttpIntegrationFixtureArgumentSets(dataStores: DataStore.SqlServer)]

Reply:
this is cool, thanks for the comment